modules/java-modules/hdfs/src/main/java/org/syslog_ng/hdfs/HdfsDestination.java

```diff
@@ -137,45 +222,72 @@ private void setKerberosAuth(Configuration configuration) throws IOException {
         UserGroupInformation.loginUserFromKeytab(options.getKerberosPrincipal(), options.getKerberosKeytabFile());
     }
 
+
+    protected void updateFilesTimer(HdfsFile hdfsfile) {
+        if (options.getTimeReap() == 0)
+           return;
+
+        if (timeReap != null)
+           return;
+
+        timeReap = new Timer();
+        timeReap.schedule( new TimeReapTask(), options.getTimeReap());
+    }
+
     @Override
     public boolean send(LogMessage logMessage) {
         isOpened = false;
         String resolvedFileName = options.getFileNameTemplate().getResolvedString(logMessage);
-        FSDataOutputStream fsDataOutputStream = getFSDataOutputStream(resolvedFileName);
-        if (fsDataOutputStream == null) {
+        lock.lock();
+        HdfsFile hdfsfile = getHdfsFile(resolvedFileName);
+        if (hdfsfile == null) {
             // Unable to open file
-            closeAll(archiveDir != null);
+            closeAll(true);
```

Comment:
hm... seems you want to always archive regardless if it is asked or not

Reply:
I think the logic here was too complex, the `archiveDir` has been checked twice once as a parameter and once later in the `archiveFiles` function.
Actually it does not make a difference if you pass `true` or `false` if you want to `archive` if archiving is configured. (see even the sentence is too complicated.)

Reply:
in that case just call closeAll(), remove the parameter and decide about archiving within this function

Reply:
The `archiveDir != null` to `true` change was functionally equivalent change. (refactor)
What you are suggestion is not, and that's why I think that should be a in a different PR.
At least I do not want to overload this PR anymore.
(It already has a leak and hdfs fix.)